### PR TITLE
Biter Battles V2 - Training Mode

### PR DESCRIPTION
**Pull request by muppet9010**
Training mode to allow 1 or more players to train against themselves when the new config option "training_mode" is set to true.
They feed their own biters.
North team bleeds evo and threat gain if any player is on the north team and the players aren't frozen (ignores requirement or >0 team south players). No threat/evo gain on south team so no biters will ever spawn for them.
Disabled South team from options and GUIs to avoid confusion. 
Team balance and Lobby minimum player counts not active when training mode is enabled.

### CODE REVIEW

**Comment by muppet9010** [comment]
I looked to minimise code changes and code overhead by removing south from lists or amending them.

**Comment by M3wM3w** [comment]
Let's instead remove the ghost-spectate button in the team manager,
with a training-mode button.

Pushing it toggles the mode on/off.

When turned on:
Teams will feed themselves instead of the enemy.
Empty teams will not get sent any attacking groups.

I think this should work out nicely.

**Comment by muppet9010** [comment]
sounds good.
I appreciate that these pull requests are as much suggestions that I have tried to work through in code to confirm validity, rather than just posting in discord.

**Comment by M3wM3w** [comment]
Oh, you can try go for it if you want!

**Comment by muppet9010** [comment]
I need to test in local MP another day as I am not sure the lobby manipulation will behave as desired when turning on & off for other players.

**Comment by muppet9010** [comment]
tested in MP with 2 players and it the lobby feels correct to me as do the changes feel to spec.
tried with the training mode turned off and it seems to behave correctly.

File: maps/biter_battles_v2/ai.lua [hunk 1]
```diff
@@ -256,15 +256,18 @@ end
...
 	local surface = game.surfaces["biter_battles"]
 	local force_name = global.next_attack
-	local biter_force_name = force_name .. "_biters"
-	local wave_amount = math.ceil(get_threat_ratio(biter_force_name) * 7)
 	
-	set_biter_raffle_table(surface, biter_force_name)
-	
-	for c = 1, wave_amount, 1 do		
-		create_attack_group(surface, force_name, biter_force_name)
+	if not global.training_mode or (global.training_mode and #game.forces[force_name].connected_players > 0) then
```
Comment (muppet9010) [inline maps/biter_battles_v2/ai.lua]: this IF wrapper being added around the existing code is the actual change. Github is just being special showing so many lines as being modified.

**Pull request merged**